provisioning/device/devdoc/transport_state_machine_requirements.md

```diff
@@ -0,0 +1,81 @@
+# transport_state_machine Requirements
+
+## Overview
+This module provides HTTP protocol support to communicate with the Azure device provisioning service
+
+## Example Usage
+``js
+  // this is effectively an abstract class that transport implements are built upon.  It is not meant to be used directly.
+``
+
+## Public Interface
+
+### constructor()
+The constructor creates a TransportStateMachine object
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_001: [** The `constructor` shall accept no arguments **]**
+
+
+### connect(callback: (err?: Error) => void): void
```

Comment:
>connect [](start = 4, length = 7)

is there any reason why we'd want to connect without registering? if ultimately register and getOperationStatus can handle connection themselves, no need to connect and then do nothing, right? so maybe it shouldn't be public?

Reply:
Absolutely.  The differences in authentication between TPM/x509 and MQTT/HTTP make it necessary to remove this as a public.

(TPM over HTTP has you pass a different authentication at registration depending on the state, but x509 over MQTT required you to pass authentication during the connect phase.)

I have a fix in a different branch,


---
In reply to: [146401297](https://github.com/Azure/azure-iot-sdk-node/pull/115#discussion_r146401297) [](ancestors = 146401297)